test/ClangImporter/Inputs/custom-modules/cxx_interop.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+namespace ns {
+
+struct Basic {
+  int a;
+};
+
+}  // namespace ns
+
+ns::Basic makeA();
+
+using ns_Basic = ns::Basic;
+
+namespace nested_struct_problem {
+
+class WrapperClass {
+ public:
+  struct NestedStruct {
```

Comment:
Can you add small comment as to why this is a problem? Say "Referring to NestedStruct before WrapperClass can trigger an infinite loop in ImportDecl" or something along the lines?